functions/database/database.js

```diff
@@ -0,0 +1,316 @@
+const admin = require("firebase-admin");
+const db = admin.firestore();
+const {Timestamp} = require("firebase-admin/firestore");
+const RELEASE_STATES = require("../utils/releaseStates.js");
+const {
+  validateNewReleaseStructure,
+  validateRelease,
+} = require("../validation/validation.js");
+const {parseCommitTitleFromMessage} = require("../utils/utils.js");
+const {REPO_URL} = require("../github/github.js");
+
+/**
+ * Retrieve the release ID of the Firestore release document
+ * with a given release name.
+ *
+ * @param {string} releaseName The release name of the release
+ * document to fetch.
+ * @throws {Error} If there is not exactly one release with the given name.
+ * @return {Promise<string>} A promise that resolves to the release ID.
+ */
+async function getReleaseID(releaseName) {
+  const releaseSnapshot = await db.collection("releases")
+      .where("releaseName", "==", releaseName)
+      .get();
+
+  if (releaseSnapshot.empty) {
+    throw new Error(`No release found with name: ${releaseName}`);
+  }
+
+  // There should only be one release with a given name
+  if (releaseSnapshot.size > 1) {
+    throw new Error(`${releaseSnapshot.size} releases found with name:
+     ${releaseName}`);
+  }
+
+  const releaseId = releaseSnapshot.docs[0].id;
+
+  return releaseId;
+}
+
+/**
+ * Retrieves release data from the database given a release ID.
+ *
+ * @param {string} releaseId The ID of the release to fetch.
+ * @return {Promise<Object>} A promise that resolves to the release data.
+ */
+async function getReleaseData(releaseId) {
+  const releaseSnapshot = await db.collection("releases")
+      .doc(releaseId)
+      .get();
+
+  if (!releaseSnapshot.exists) {
+    throw new Error(`No release found with ID: ${releaseId}`);
+  }
+
+  const releaseData = releaseSnapshot.data();
+
+  return releaseData;
+}
+
+/** Retrieve all past releases from Firestore. This includes releases that are
+ * currently active.
+ *
+ * @return {Promise<Array>} Returns a promise that resolves to an array
+ * of Firestore document data of past releases.
+ * @throws {Error} If there is an error in fetching data from Firestore,
+ * the promise is rejected with an error.
+ */
+async function getPreviousReleaseData() {
+  const today = new Date();
+  const snapshot = await db.collection("releases")
+      .where("releaseDate", "<", today)
+      .get();
+
+  const pastReleases = snapshot.docs.map((doc) => doc.data());
+
+  return pastReleases;
+}
+
+/** Write new releases to Firestore. This function assumes that the contents
+ * of the releases are validated. newReleases is an array of release objects
+ * that have the following structure:
+ * {
+ *   releaseName: string,
+ *   releaseOperator: string,
+ *   codeFreezeDate: Date,
+ *   releaseDate: Date,
+ * }
+ *
+ * @param {Object} newReleases - Releases to store in Firestore
+ */
+async function addReleases(newReleases) {
+  // Validate the structure of the new releases before we store
```

Comment:
No need for comment, the method name and docs on the method itself are more than enough.

Reply:
Resolved in https://github.com/firebase/firebase-release-dashboard/pull/8/commits/a32e049b16404c7818f488231642a11b343b3068